### PR TITLE
[WIP] HtmlViewer- Create new component (and sub-components) and a basic story

### PR DESCRIPTION
**Pull request by ColbyJohnIBM**
#### What do these changes do/fix?

Add a component to be used for CI documents, enrichments view, and simple document view (document preview).

https://github.ibm.com/Watson-Discovery/the-tooling-core-tracker/issues/1259

#### How do you test/verify these changes?

Open Storybook and look at the HtmlViewer story. Click through the knobs to see different types of documents.

#### Have you documented your changes (if necessary)?

Yes, in Storybook.

#### Are there any breaking changes included in this pull request?

No.

### CODE REVIEW

**Comment by ColbyJohnIBM** [comment]
This PR still needs quite a few changes / additions, including:
[ ] Handle simple document view
[ ] Handle `TODO` items
[ ] Fix typing bugs (and remove uses of `any`)
[ ] Add unit tests
[ ] Clean up fixtures
[ ] Add in state overrides for CI documents (although maybe this should be done as part of https://github.ibm.com/Watson-Discovery/the-tooling-core-tracker/issues/1260)

**Comment by H-Fin** [comment]
Pulled it down and opened up Storybook, its looking good so far! Twice now when I click on `parse error` under `HtmlViewer` is causes Storybook to become unresponsive for me.

**Comment by ColbyJohnIBM** [comment]
Ah, good call, I hadn't looked at the parse error story in a bit-- I'll look into that...

**Comment by broulaye** [comment]
Question, My understanding was this is suppose to replace the html view that's currently there. correct?

File: packages/discovery-react-components/src/components/HtmlViewer/HtmlViewer.tsx [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+  theme?: Theme;
+  /**
+   * Override autosizing of document content with specified width. Useful for testing.
+   */
+  overrideDocWidth?: number;
+  /**
+   * Override autosizing of document content with specified height. Useful for testing.
+   */
+  overrideDocHeight?: number;
+}
```
Comment (broulaye) [inline packages/discovery-react-components/src/components/HtmlViewer/HtmlViewer.tsx]: this is missing `didCatch`

File: packages/discovery-react-components/src/components/HtmlViewer/HtmlViewer.tsx [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+   * Specify which field in the document holds the text data to display
+   */
+  dataField?: string;
+  /**
+   * List of ids of elements to highight
+   */
+  // TODO: colbyj restrict this to string array (ids)?
+  highlightedList: any[];
+  /**
+   * Callback function on document processing success
```
Comment (H-Fin) [inline packages/discovery-react-components/src/components/HtmlViewer/HtmlViewer.tsx:80]: I believe this needs to be switched with the description below.

Comment (ColbyJohnIBM) [reply packages/discovery-react-components/src/components/HtmlViewer/HtmlViewer.tsx:80]: Oops. 🤦‍♂

**Comment by maniax89** [comment]
@ColbyJohnIBM is it safe to close this PR?

**Comment by ColbyJohnIBM** [comment]
😢 Yeah, I guess...

(If we re-prioritize refactoring this, we can use this closed PR for reference.)

**Pull request closed without merging**